src/builtins/core/date.rs

```diff
@@ -600,8 +600,52 @@ impl PlainDate {
             .with_calendar(self.calendar.identifier(), display_calendar)
             .build()
     }
+
+    #[inline]
+    pub fn to_zoned_date_time(
```

Comment:
issue: provider APIs should use the `with_provider` suffix -> `to_zoned_date_time_with_provider`

Also, since this is a provider API, a nonprovider API can be added to `src/builtins/compiled/date.rs` if you're comfortable with doing so (I think this may be the first provider API for date, so the module may need to be created for it).